src/api.rs

```diff
@@ -75,45 +119,82 @@ pub const PAGE_SHIFT: u32 = 12;
 pub const PAGE_SIZE: u32 = 4096;
 
 pub trait Introspectable {
-    // get VCPU count
+    /// Get number of vcpus
+    ///
+    /// It does not take any parameter and returns the number of vcpus in 16 bit unsigned integer format.
+    ///
     fn get_vcpu_count(&self) -> Result<u16, Box<dyn Error>> {
         unimplemented!();
     }
 
-    // read physical memory
+    /// Used to read the contents of physical memory.
+    ///
+    /// # Arguments
+    ///
+    /// * 'paddr' - the physical address of memory from where we are to read
+    /// * 'buf' - pointer to an array of suitable size where the contents thus read are to be dumped
+    ///
```

Comment:
read the physical memory, starting from `paddr`, into `buf`

* 'paddr' - the physical address to read from
* `buf` - the data read from memory